### PR TITLE
clicking thumbnail in the playlist menu should cause video to play

### PR DESCRIPTION
**Pull request by saxena-gaurav**

### CODE REVIEW

File: test/videojs-playlist-ui.test.js [hunk 1]
```diff
@@ -374,6 +374,7 @@ test('tracks when an ad is playing', function() {
 test('changes the selection when tapped', function(test) {
   player.playlist(playlist);
   player.playlistUi();
+  player.play = function(){};
```
Comment (bclwhitaker) [inline test/videojs-playlist-ui.test.js]: This should actually test that play was called.

File: lib/videojs-playlist-ui.js [hunk 2]
```diff
@@ -81,6 +81,7 @@ videojs.PlaylistMenuItem = videojs.Component.extend({
 
     this.on(['click', 'tap'], (event) => {
       player.playlist.currentItem(indexOf(player.playlist(), this.item));
+      player.play();
```
Comment (bclwhitaker) [inline lib/videojs-playlist-ui.js]: If someone was depending on the previous behavior this might suck.

**Comment by gkatsev** [comment]
I feel like this needs to be feature-switched so you can turn it on or off in the plugin's options.

File: lib/videojs-playlist-ui.js (hunk 2 above)
Comment (gkatsev) [inline lib/videojs-playlist-ui.js]: That's fine, we could always just release it as a 2.0

**Comment by dmlap** [comment]
+1 to offering an option to turn on this behavior.

File: lib/videojs-playlist-ui.js [hunk 3]
```diff
@@ -67,7 +67,7 @@ const createThumbnail = function(thumbnail) {
 };
 
 videojs.PlaylistMenuItem = videojs.Component.extend({
-  init: function(player, options) {
+  init: function(player, options, settings) {
```
Comment (dmlap) [inline lib/videojs-playlist-ui.js]: Why isn't the new setting included in `options`?

File: test/videojs-playlist-ui.test.js [hunk 4]
```diff
@@ -372,8 +372,12 @@ test('tracks when an ad is playing', function() {
 // -----------
 
 test('changes the selection when tapped', function(test) {
+  var playCalled = false;
   player.playlist(playlist);
-  player.playlistUi();
+  player.playlistUi({'playOnClick': true});
```
Comment (dmlap) [inline test/videojs-playlist-ui.test.js]: I think we should keep a test where this new setting is false so we verify both variations work properly.

File: README.md [hunk 5]
```diff
@@ -49,6 +49,12 @@ Default: "vjs-playlist"
 
 The name of the class to search for to populate the playlist menu.
 
+#### playOnClick
```
Comment (dmlap) [inline README.md]: Naming nitpick: click is pretty mouse-specific. What about `playOnSelect` to account for touch interactions?

File: lib/videojs-playlist-ui.js (hunk 3 above)
Comment (saxena-gaurav) [inline lib/videojs-playlist-ui.js]: The `options` argument here is the playlist item object containing `description`, `name` and `duration`, where as `settings` argument here is the "options" to the plugin - `playOnSelect`, `className`. Should we rename `options` argument to `playlistItem` ?

File: test/videojs-playlist-ui.test.js (hunk 4 above)
Comment (saxena-gaurav) [inline test/videojs-playlist-ui.test.js]: done.

File: lib/videojs-playlist-ui.js (hunk 3 above)
Comment (dmlap) [inline lib/videojs-playlist-ui.js]: Yeah, that sounds like a better name to me.

**Comment by dmlap** [comment]
LGTM

**Pull request merged**